rpcs3/Emu/RSX/GL/rsx_gl_texture.cpp

```diff
@@ -406,31 +406,58 @@ namespace rsx
 
 		void texture::init(int index, rsx::texture& tex)
 		{
+			switch (tex.dimension())
+			{
+			case rsx::texture_dimension::dimension3d:
+				if (!tex.depth())
```

Comment:
Can't this `if` be factored out of switch? Also, no `break` in first two cases, doesn't seem to be intended.